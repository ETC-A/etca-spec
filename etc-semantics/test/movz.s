;-mstrict
;0,16,80,144
; test that movz works under each mode. ketc will ignore writes to exten bits
; that aren't supported by the cpuid. movz under cpuid 0 should fail unsupported.
; We test movzh and movzx under byte_ops, byte_ops+dops, and byte_ops+qops

    movx  %rx0,%exten
    movx  %rx1,0xd0   ; byte ops, dops, qops
    orx   %rx0,%rx1
    movx  %exten,%rx0 ; and enable them (byte ops were already enabled)
.extension byte_operations
    movzh %rh0,-1     ; %r0 <= 255
    movzx %rx1,-1     ; %r0 <= 65535
end:
    jmp   end
