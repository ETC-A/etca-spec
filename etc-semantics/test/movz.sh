set -u
./scripts/test.sh "$1" "$2" "$3" "test/movz.bin" 0
./scripts/test.sh "$1" "$2" "$3" "test/movz.bin" 16
./scripts/test.sh "$1" "$2" "$3" "test/movz.bin" 80
./scripts/test.sh "$1" "$2" "$3" "test/movz.bin" 144
