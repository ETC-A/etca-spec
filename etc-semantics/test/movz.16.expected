<etc>
  <k>
    #halt "Halted successfully!" ~> .
  </k>
  <pc> 32782 </pc>
  <registers>
    0 |-> maxUInt8
    1 |-> maxUInt16
    2 |-> 0
    3 |-> 0
    4 |-> 0
    5 |-> 0
    6 |-> 0
    7 |-> 0
  </registers>
  <memory>
    128 |-> b"^\x02Y&\\0\x14\x04_\x02H\x1fX?\x8e\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
  </memory>
  <flags>
    <carry>    false </carry>
    <zero>     false </zero>
    <negative> false </negative>
    <overflow> false </overflow>
  </flags>
  <machine-details>
    <cpuid> 16 </cpuid>
    <exten> 16 </exten>
    <reg-mode>  word </reg-mode>
    <reg-width> word </reg-width>
    <reg-count> 8 </reg-count>
    <evil-mode> false </evil-mode>
  </machine-details>
  <in> </in>
  <out> </out>
</etc>
